### DISCUSSION TITLE
Apparently no way to set log level to "warning"

### DISCUSSION THREAD

**Discussion by gwhitney** [mkdocs/mkdocs#3834] [General]
There are command line arguments that set the log level to "debug" (which is super verbose) and "error" (which generally produces nothing unless something pretty bad is going on). But some plugins produce _lots_ of info messages, so the default "info" level overwhelms any useful information. And we run with --strict, but setting the log level to "error" disables warnings, and so if we set that we would get no benefit at all from "strict". (It came as quite a surprise to us that changing the _log level_ would influence whether a doc build completes successfully, but that is the case when using --strict.)

So for us the ideal log level would be warning, but I can't see any way to set that. I'd suggest a cli option `--log-level` that would accept any level (so that -v and -q are just shorthands for `--log-level debug` and `--log-level error`).

One potential drawback of log level = warning is that for `mkdocs serve`, the link and list of files being watched is an info message, as are the access logs, and those can be useful. So I would go so far as to suggest a separate `--server-level` option, to give a log level that is switched to just before the link, etc. is emitted.

If you would like a PR along these lines please just let me know what options you would like implemented, and I will be happy to supply a PR.

**Comment by lovelydinosaur** [comment]
Perfectly reasonable suggestion, do we do need to balance dealing-with-all-feature-requests vs minimizing configuration options.

**Reply by gwhitney** [reply]
I totally sympathize with that point. If we had just the --log-level argument, that would already be a help, and in some sense it actually reduces the number of options, as it covers two existing ones that can then just be thought of as backwards-compatibility abbreviations.
